compiler/src/iree/compiler/Codegen/Common/DecomposeConvolutionToLowerDimOps.cpp

```diff
@@ -16,6 +19,106 @@ namespace mlir::iree_compiler {
 
 namespace {
 
+static bool foldHDim(linalg::DepthwiseConv2DNhwcHwcOp convOp) {
+  Value kernel = convOp.getInputs().back();
+  Value output = convOp.getOutputs().front();
+
+  auto kernelType = dyn_cast<RankedTensorType>(kernel.getType());
+  auto outputType = dyn_cast<RankedTensorType>(output.getType());
+
+  auto kernelShape = kernelType.getShape();
+  auto outputShape = outputType.getShape();
+
+  int64_t khSize = kernelShape[0];
+  int64_t ohSize = outputShape[1];
+  bool removeH = (khSize == 1 && ohSize == 1);
+
+  return removeH;
+}
+
+/// Computes a "decomposed" lowering config attribute for a conv OP
+///
+/// This method complements the patterns to decompose 2D convolutions into 1D
+/// convs. Specifically, it will update the lowering config attached to a Conv
+/// Op in a way that matches the "decomposition" patterns.
+///
+/// At the moment only Depthwise HWC convolutions are supported.
+static FailureOr<IREE::Codegen::LoweringConfigAttr>
+computeNewLCA(ArrayRef<Operation *> computeOps, MLIRContext *context) {
+
+  // 0.1 Double-check that there's only one convolution Op.
+  // TODO: Make this hook work with multiple conv Ops
+  assert(llvm::count_if(computeOps,
+                        [](Operation *op) {
+                          return isa<linalg::ConvolutionOpInterface>(op);
+                        }) == 1 &&
+         "Exactly 1 Linalg Conv Op is expected");
+
+  // 0.2 ATM only 2D depthwise HWC convs are supported.
+  // TODO: Add support for other convs
+  bool seen2DDepthwiseConvHWC = false;
+  linalg::DepthwiseConv2DNhwcHwcOp convOp;
+  for (auto op : computeOps) {
+    if (isa<linalg::DepthwiseConv2DNhwcHwcOp>(op)) {
+      seen2DDepthwiseConvHWC = true;
+      convOp = cast<linalg::DepthwiseConv2DNhwcHwcOp>(op);
+      break;
+    }
+  }
+  if (!seen2DDepthwiseConvHWC)
+    return failure();
+
+  // 0.3. ATM only folding of the H dim is supported.
+  // TODO: Add support for cases where the W dim is folded.
+  if (!foldHDim(convOp))
+    return failure();
+
+  // 1. Get the current lowering config attached to the Conv Op.
+  FailureOr<IREE::Codegen::LoweringConfigAttr> loweringConfigAttr =
+      getLoweringConfig(computeOps);
+  if (failed(loweringConfigAttr))
+    return failure();
+
+  TilingConfig tc(loweringConfigAttr.value());
+
+  // 2. Calculate new tiling levels.
+  // Note that this will basically erase the _H_ dims from the orignal lowering
+  // config.
+  SmallVector<unsigned> hDimsToErase{1, 4};
```

Comment:
`linalg::classifyConvolutionDims` can be used to retrieve these indices once extending to more convolution types.

Reply:
I think it is `linalg::inferConvolutionDims` method. I was not aware of it, it looks helpful!

https://github.com/llvm/llvm-project/blob/ab22504479cb7b8985855aa2f2c851390bc1d167/mlir/include/mlir/Dialect/Linalg/IR/LinalgInterfaces.h#L85-L109